app/models/production_batch.rb

```diff
@@ -15,6 +15,16 @@ def check_batch_name
     errors.add(:name, :taken) if u > 0
   end
 
+  def distributor_inc_vat
+    return 0 if !distributor_price || !vat
+    (distributor_price + distributor_price * vat.add_factor)/100
+  end
+
+  def retail_inc_vat
+    return 0 if !retail_price || !vat
+    (retail_price + retail_price * vat.add_factor)/100
```

Comment:
division by 100 here... If this is to show the price I think these (`*_inc_vat`) methods belongs be in the decorator and not in the "model".